packages/core/js-client/src/compilerSupport/callFunction.ts

```diff
@@ -51,95 +45,63 @@ const log = logger("aqua");
  * @returns
  */
 
-type CallAquaFunctionArgs = {
-  def: FunctionCallDef;
+export type CallAquaFunctionArgs = {
   script: string;
-  config: FnConfig;
+  config: CallAquaFunctionConfig | undefined;
   peer: FluencePeer;
-  args: PassedArgs;
+  args: { [key: string]: JSONValue | ArgCallbackFunction };
+  fireAndForget: boolean | undefined;
+};
+
+export type CallAquaFunctionConfig = {
+  ttl?: number;
 };
 
 export const callAquaFunction = async ({
-  def,
   script,
-  config,
+  config = {},
   peer,
   args,
+  // TODO: remove after LNG-286 is done
+  fireAndForget = false,
 }: CallAquaFunctionArgs) => {
-  // TODO: this function should be rewritten. We can remove asserts if we wont check definition there
-  log.trace("calling aqua function %j", { def, script, config, args });
-  const argumentTypes = getArgumentTypes(def);
+  log.trace("calling aqua function %j", { script, config, args });
 
   const particle = await peer.internals.createNewParticle(script, config.ttl);
 
-  return new Promise((resolve, reject) => {
+  return new Promise<JSONValue>((resolve, reject) => {
     for (const [name, argVal] of Object.entries(args)) {
```

Comment:
1. What happens in this loop? Some doc would be nice
2. WDYT, maybe move this loop to a separate function? `callAquaFunction` is pretty big

Reply:
1. will add, pls check
2. I refactored function and made it smaller